### PR TITLE
BUG: Right result for DatetimeIndex + TimeDelta when timezone is set(…

### LINKED ISSUE #13905: DatetimeIndex + TimeDelta gives wrong results when timezone is set
When I add a time delta to a DatetimeIndex which is tz-aware, the results appear as UTC, but sometimes claim to be in the original non-UTC timezone.
#### Code

```
import pandas as pd
index = pd.DatetimeIndex(['2016-06-28 05:30', '2016-06-28 05:31'], dtype='datetime64[ns, America/Chicago]')
offset = pd.Series(['00:00:05', '00:00:05'], dtype='timedelta64[ns]', index=index)
print(index + offset)
print(index + offset.values)
```
#### Actual Output

```
2016-06-28 05:30:00-05:00   2016-06-28 10:30:05
2016-06-28 05:31:00-05:00   2016-06-28 10:31:05
dtype: datetime64[ns]
DatetimeIndex(['2016-06-28 10:30:05-05:00', '2016-06-28 10:31:05-05:00'], dtype='datetime64[ns, America/Chicago]', freq=None)
```
#### Expected Output

```
2016-06-28 05:30:00-05:00   2016-06-28 05:30:05-05:00
2016-06-28 05:31:00-05:00   2016-06-28 05:31:05-05:00
dtype: datetime64[ns]
DatetimeIndex(['2016-06-28 05:30:05-05:00', '2016-06-28 05:31:05-05:00'], dtype='datetime64[ns, America/Chicago]', freq=None)
```

Note however that this does print the expected result:

```
print(index + pd.to_timedelta('00:00:05'))
```

I'm using Pandas 0.18.1.

### PR DESCRIPTION
**Pull request by conquistador1492**
- [x] closes #13905 
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

…#13905)

### CODE REVIEW

File: pandas/core/ops.py [hunk 1]
```diff
@@ -461,7 +461,7 @@ def _convert_to_array(self, values, name=None, other=None):
 
             # a datelike
             elif isinstance(values, pd.DatetimeIndex):
-                values = values.to_series()
```
Comment (jreback) [inline pandas/core/ops.py]: we should handle ``is_datetime64_dtype`` and ``is_datetime64tz_dtype`` separately here

File: pandas/indexes/base.py [hunk 2]
```diff
@@ -165,7 +165,15 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
             if (issubclass(data.dtype.type, np.datetime64) or
                     is_datetimetz(data)):
                 from pandas.tseries.index import DatetimeIndex
-                result = DatetimeIndex(data, copy=copy, name=name, **kwargs)
+                if 'tz' in kwargs:
```
Comment (jreback) [inline pandas/indexes/base.py]: this is way too complicated

File: pandas/tests/series/test_operators.py [hunk 3]
```diff
@@ -373,6 +373,16 @@ def test_timedelta64_operations_with_DateOffset(self):
             td - op(5)
             op(5) - td
 
+    def test_timedelta64_operations_with_DatetimeIndex_tz(self):
+        dti = pd.DatetimeIndex(['2016-06-28 05:30', '2016-06-28 05:31'],
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: add the issue number as a comment

File: pandas/core/ops.py [hunk 4]
```diff
@@ -543,9 +543,9 @@ def _offset(lvalues, rvalues):
 
                 # with tz, convert to UTC
                 if self.is_datetime64tz_lhs:
-                    lvalues = lvalues.tz_localize(None)
```
Comment (sinhrks) [inline pandas/core/ops.py]: I *think* this localization can be removed, as adding timedelta is unrelated to the timezone.

File: pandas/indexes/base.py (hunk 2 above)
Comment (sinhrks) [inline pandas/indexes/base.py]: we can use ``DatetimeIndex._simple_new(data, name=name, **kwargs)``. Needs to handle copy in advance.

**Comment by sinhrks** [comment]
Pls add a test for ``DatetimeIndex`` constructor covering possible patterns.

**Comment by jreback** [comment]
acutally need to localize if freq < 1D, otherwise can simply add it.

E.g. if we are crossing a dst boundary, then need to convert to UTC, operate, then re-convert.

So I think we should continue to always convert / operate / re-convert for simplicity.

**Comment by sinhrks** [comment]
@jreback you're right, but ``DatetimeIndex`` itself handles this, so not needed in ``ops``?

- https://github.com/pydata/pandas/blob/master/pandas/tseries/index.py#L771

File: pandas/core/ops.py (hunk 1 above)
Comment (conquistador1492) [inline pandas/core/ops.py]: it happens, isn't is?

**Comment by jreback** [comment]
can you rebase / update?

**Comment by jreback** [comment]
can you rebase / update

**Comment by jreback** [comment]
can you rebase / update

**Comment by jreback** [comment]
closing, but if you want to update, pls comment

**Pull request closed without merging**